content/recipes/swagger.md

```diff
@@ -77,75 +73,179 @@ async create(@Body() createCatDto: CreateCatDto) {
 
 <figure><img src="/assets/swagger-dto.png" /></figure>
 
-As you can see, the definition is empty although the class has a few declared properties. In order to make the class properties accessible to the `SwaggerModule`, we have to mark all of them with `@ApiModelProperty()` decorator:
+As you can see, the definition is empty although the class has a few declared properties. In order to make the class properties visible to the `SwaggerModule`, we have to either annotate them with `@ApiProperty()` decorator or use a CLI plugin (read more in the **Plugin** section):
 
 ```typescript
-import { ApiModelProperty } from '@nestjs/swagger';
+import { ApiProperty } from '@nestjs/swagger';
 
 export class CreateCatDto {
-  @ApiModelProperty()
-  readonly name: string;
+  @ApiProperty()
+  name: string;
 
-  @ApiModelProperty()
-  readonly age: number;
+  @ApiProperty()
+  age: number;
 
-  @ApiModelProperty()
-  readonly breed: string;
+  @ApiProperty()
+  breed: string;
 }
 ```
 
 Let's open the browser and verify the generated `CreateCatDto` model:
 
 <figure><img src="/assets/swagger-dto2.png" /></figure>
 
-The `@ApiModelProperty()` decorator accepts the following options object:
+In addition, the `@ApiProperty()` decorator allows setting various [Schema Object](https://swagger.io/specification/#schemaObject) properties:
 
 ```typescript
-export const ApiModelProperty: (metadata?: {
-  description?: string;
-  required?: boolean;
-  type?: any;
-  isArray?: boolean;
-  collectionFormat?: string;
-  default?: any;
-  enum?: SwaggerEnumType;
-  format?: string;
-  multipleOf?: number;
-  maximum?: number;
-  exclusiveMaximum?: number;
-  minimum?: number;
-  exclusiveMinimum?: number;
-  maxLength?: number;
-  minLength?: number;
-  pattern?: string;
-  maxItems?: number;
-  minItems?: number;
-  uniqueItems?: boolean;
-  maxProperties?: number;
-  minProperties?: number;
-  readOnly?: boolean;
-  xml?: any;
-  example?: any;
-}) => PropertyDecorator;
-```
-
-> warning **Hint** There's an `@ApiModelPropertyOptional()` shortcut decorator which helps to avoid continuous typing `{{"@ApiModelProperty({ required: false })"}}`.
-
-Thanks to that we can simply set the **default** value, determine whether the property is required or explicitly set the type.
+@ApiProperty({
+  description: 'The age of a cat',
+  min: 1,
+  default: 1,
+})
+age: number;
+```
+
+> info **Hint** Instead of explicitly typing `{{"@ApiProperty({ required: false })"}}` you can use `@ApiPropertyOptional()` short-hand decorator.
+
+In order to explicitly set the type of the property, use `type` key:
+
+```typescript
+@ApiProperty({
+  type: Number,
+})
+age: number;
+```
+
+#### Enums
+
+To be able for `SwaggerModule` to identify an `enum`, we have to manually set the `enum` property on `@ApiProperty` with an array of values.
+
+```typescript
+@ApiProperty({ enum: ['Admin', 'Moderator', 'User']})
+role: UserRole;
+```
+
+You can also define a real `UserRole` enum instead as following:
```

Comment:
You can also define a real `UserRole` enum instead as following:

->

Alternatively, define an actual TypeScript enum  as follows: